src/variables.ts

```diff
@@ -17,15 +17,19 @@ export class SitewiseVariableSupport extends CustomVariableSupport<DataSource, S
   editor = QueryEditor;
 
   query(request: DataQueryRequest<SitewiseQuery>): Observable<DataQueryResponse> {
-    assign(request.targets, [{ ...request.targets[0], refId: 'A' }]);
-    const response = this.datasource.query(request);
-    switch (request.targets[0].queryType) {
-      case QueryType.ListAssetModels:
-      case QueryType.ListAssets:
-      case QueryType.ListAssociatedAssets:
-        return this.parseOptions(response);
-      default:
-        return response;
+    if (this.isValidQuery(request.targets[0])) {
```

Comment:
Nice! This should prevent unnecessary invocation of `query(...)`.
Curious if there's another place/interface for handling query validations?

Reply:
There isn't one for queries in template variable editor. For normal queries there's `filterQuery` method [here](https://github.com/grafana/grafana/blob/763f0bac90d6296f034939627e9058bf2071b484/packages/grafana-data/src/types/datasource.ts#L276); however this quietly filters out the invalid query so it's normally only used for datasources where onChange triggers query request. In Sitewise, we've decided to go the way of not triggering a request automatically, so we don't really need the `filterQuery` yet.